front/src/modules/timesStops/helpers/__tests__/utils.spec.ts

```diff
@@ -0,0 +1,503 @@
+        departure: { time: '23:50:00' },
+        stopFor: '300', // no longer correct, not yet updated by the function
+      } as PathWaypointRow;
+      const previousRowData = {
+        opId: 'd94a2af4',
+        name: 'Gr',
+        arrival: { time: '23:40:00' },
+        departure: { time: '23:45:00' },
+        stopFor: '300',
+      } as PathWaypointRow;
+      const result = updateRowTimesAndMargin(rowData, previousRowData, whateverOperation, 4);
+      expect(result).toEqual({
+        opId: 'd94a2af4',
+        name: 'Gr',
+        arrival: { time: '23:40:00' },
+        departure: { time: '23:50:00' },
+        stopFor: '600', // now correct with the new arrival and departure
+        isMarginValid: true,
+      });
+    });
+  });
+  describe('theoritical margin is incorrect', () => {
+    it('should set isMarginValid flag to false', () => {
+      const rowData = {
+        opId: 'd94a2af4',
+        name: 'Gr',
+        theoreticalMargin: '10',
+      } as PathWaypointRow;
+      const previousRowData = {
+        opId: 'd94a2af4',
+        name: 'Gr',
+      } as PathWaypointRow;
+      const result = updateRowTimesAndMargin(rowData, previousRowData, whateverOperation, 4);
+      expect(result.isMarginValid).toBe(false);
+    });
+  });
+  describe('user removed first row theoritical margin', () => {
+    it('should set the theoritical margin back to 0%', () => {
+      const rowData = {
+        opId: 'd94a2af4',
+        name: 'Gr',
+      } as PathWaypointRow;
+      const previousRowData = {
+        opId: 'd94a2af4',
+        name: 'Gr',
+        theoreticalMargin: '10%',
+      } as PathWaypointRow;
+      const operation = {
+        fromRowIndex: 0,
+      };
+      const result = updateRowTimesAndMargin(rowData, previousRowData, operation, 4);
+      expect(result).toEqual({
+        opId: 'd94a2af4',
+        name: 'Gr',
+        arrival: undefined,
+        isMarginValid: true,
+        onStopSignal: false,
+        theoreticalMargin: '0%',
+      });
+    });
+  });
+  describe('arrival is before midnight, departure after midnight', () => {
+    it('should compute the stopFor correctly', () => {
+      const rowData = {
+        opId: 'd94a2af4',
+        name: 'Gr',
+        arrival: { time: '23:40:00' },
+        departure: { time: '00:20:00' },
+        stopFor: '300',
+      } as PathWaypointRow;
+      const previousRowData = {
+        opId: 'd94a2af4',
+        name: 'Gr',
+        arrival: { time: '23:40:00' },
+        departure: { time: '23:45:00' },
+        stopFor: '300',
+      } as PathWaypointRow;
+      const result = updateRowTimesAndMargin(rowData, previousRowData, whateverOperation, 4);
+      expect(result).toEqual({
+        opId: 'd94a2af4',
+        name: 'Gr',
+        arrival: { time: '23:40:00' },
+        departure: { time: '00:20:00' },
+        stopFor: '2400',
+        isMarginValid: true,
+      });
+    });
+  });
+});
+
+describe('updateTimeAndDays', () => {
```

Comment:
Maybe it's better to put the name of the function here like we usually do

```suggestion
describe('updateDaySinceDeparture', () => {
```

Reply:
yes, that was forgotten when function was renamed